command/agent/http.go

```diff
@@ -531,13 +531,7 @@ func (s *HTTPServer) parseToken(req *http.Request, token *string) {
 // DC in the request, if given, or else the agent's DC.
 func (s *HTTPServer) parseSource(req *http.Request, source *structs.QuerySource) {
 	s.parseDC(req, &source.Datacenter)
-	if node := req.URL.Query().Get("near"); node != "" {
```

Comment:
I think this'll break other endpoints that depend on `?near=_agent` getting resolved at this level. I think you can pass the existing `Source` member of the execute request into here.